remove DRAM temporal reduction check

nellie-wu opened the pull request.

angshuman-parashar reviewed (changes requested):
I actually think the user-facing reduction_supported attibute is a good idea. We'll need it eventually. The only thing I suggest removing is the check in lines 1183-1190.

nellie-wu commented:
I reverted the interface and updated the specification for reduction from `yes` string to `true` bool, as I feel that's more intuitive. Let me know if the previous one is better, or I should keep both for compatibility.
